kubernetes/kubernetes.go

```diff
@@ -217,15 +226,28 @@ func CollectKubeMetrics(config KubeAgentConfig) {
 
 }
 
+// isCustomUploadEnvsSet checks the KubeAgentConfig for custom upload ENVs and returns true/false and the provider mode
+// Currently supported: AWS and Azure, the goal is to add more support with this function later
+func isCustomUploadEnvsSet(ka *KubeAgentConfig) (bool, string) {
+	var isCustomUpload bool
+	var agentMode string
+	isCustomUpload, agentMode = isCustomS3UploadEnvsSet(ka)
+	if isCustomUpload {
+		return isCustomUpload, agentMode
+	}
+	isCustomUpload, agentMode = isCustomAzureUploadEnvsSet(ka)
+	return isCustomUpload, agentMode
+}
+
 // isCustomS3UploadEnvsSet checks to see if the agent has a custom S3 location and S3 region to upload to
 // if both these variables are not set, default upload to Apptio S3
-func isCustomS3UploadEnvsSet(ka *KubeAgentConfig) bool {
+func isCustomS3UploadEnvsSet(ka *KubeAgentConfig) (bool, string) {
```

Comment:
String return here is unnecessary imo, could consolidate this into `isCustomUploadEnvsSet`. Alternatively an enum or three consts could satisfy the requirements of both vars.

isCustomS3Upload := isCustomS3UploadEnvsSet(ka)
	if isCustomS3Upload {
		return true, aws
	}
	isCustomAzureUpload := isCustomAzureUploadEnvsSet(ka)
	if isCustomAzureUpload {
		return true, azure
	}
	return false,""